lib/src/request_processor.cc

```diff
@@ -800,11 +826,23 @@ void RequestProcessor::loadModulesConfiguration()
     }
 }
 
-void RequestProcessor::loadInternalModules()
+void RequestProcessor::registerModule(Module *module)
+{
+    std::shared_ptr<Module> module_ptr { module };
+
+    if (!modules_.emplace(module_ptr->module_name, module_ptr).second) {
+        LOG_WARNING("Ignoring attempt to re-register module: {1}", module_ptr->module_name);
+    }
+}
+
+void RequestProcessor::loadInternalModules(const Configuration::Agent& agent_configuration)
 {
-    // HERE(ale): no external configuration for internal modules
-    modules_["echo"] = std::shared_ptr<Module>(new Modules::Echo);
-    modules_["ping"] = std::shared_ptr<Module>(new Modules::Ping);
+    registerModule(new Modules::Echo);
+    registerModule(new Modules::Ping);
+    // TODO(task) use agent_configuration.task_cache_dir instead of system_prefix
+    registerModule(new Modules::Task(Configuration::Instance().getExecPrefix(),
+                                     system_prefix()+"/pxp-agent/tasks",
+                                     spool_dir_path_.string()));
```

Comment:
The 3rd argument is spool_dir. The 2nd is what the TODO covers, change from using the old path (preserved to avoid breaking existing workflows until we have file downloading working) to task_cache_dir.